### PR TITLE
Added ability to specify ellipsis location (start, middle, end)

### PR DESCRIPTION
**Pull request by jibsaramnim**
This PR adds a new property that lets you specify where you would like to have the ellipsis be shown. By default this appears at the `end`, but now you can also choose `start` and `middle`. The latter splits the text in half*, placing the ellipsis right in the middle.

I must apologize in advance as I do not speak Chinese, so I was not able to provide translations for the added content on the demo page. It would be great if you could take a look at this and, if it looks alright to you too, merge this. I have tried to make sure I followed your existing formatting, but please let me know if I made any mistakes, or missed any details.

Thank you!

_* Using `Math.floor` on the left-hand side, and `Math.ceil` on the right-hand side, to support uneven string lengths too._

### CODE REVIEW

**Review by Justineo — CHANGES_REQUESTED** [review]
And can you please remove the compiled demo from the PR? We should probably gitignore the `demo` directory as will be automatically distributed to Vercel now and doesn't rely on GitHub pages.

File: src/components/Clamp.js [hunk 1]
```diff
@@ -17,6 +17,13 @@ export default {
       type: String,
       default: '…'
     },
+    location: {
+      type: String,
+      default: 'end',
+      validator: function (value) {
+        return ['start', 'middle', 'end'].includes(value)
```
[the remark this answers is not in the document]
Comment (jibsaramnim) [reply src/components/Clamp.js]: Ah, I should've checked what browser compatibility I should have aimed for, apologies! I have applied your suggestions, thank you kindly!

**Comment by jibsaramnim** [comment]
@Justineo thank you kindly for taking a look! I have applied the changes you have suggested, including removing the compiled demo files and adding that folder to `.gitignore`. Please let me know if there is anything else you'd like me to do, I'm more than happy to help! :)

**Comment by Justineo** [comment]
Thank you!

**Comment by jibsaramnim** [comment]
@Justineo just a small note; it might be useful to update the Github repo demo URL, as it's still pointing to https://justineo.github.io/vue-clamp/demo/ which isn't working anymore now.

**Comment by Justineo** [comment]
@jibsaramnim Oops I forgot about that. Fixed now, thank you!

**Pull request merged**